common/src/main/java/org/apache/comet/parquet/NativeBatchReader.java

```diff
@@ -459,7 +459,9 @@ public void close() throws IOException {
       importer = null;
     }
     nativeUtil.close();
-    Native.closeRecordBatchReader(this.handle);
+    if (handle > 0) {
+      Native.closeRecordBatchReader(this.handle);
+    }
```

Comment:
And should we set handle to zero after being closed?

Reply:
Thanks, that makes sense to me.